Tmedia 468 header nav chain buttons

JackHowa opened the pull request:
## Description
Header nav chain buttons, querly, arc-search, sections

## Jira Ticket
- [TMEDIA-468](https://arcpublishing.atlassian.net/browse/TMEDIA-468)

## Acceptance Criteria

- Match color logic for nav color and borders for sections, queryly, arc search (not expanded)
- Expanded input should always have white background and incorporate the button similarly as before

## Test Steps

1. Checkout this branch `git checkout TMEDIA-468-header-nav-chain-buttons`
2. Run fusion repo with linked blocks `npx fusion start -f -l  @wpmedia/identity-block,@wpmedia/shared-styles,@wpmedia/header-nav-chain-block`
3. Create homepage with sections, queryly on the left and arc search on the right area
4. Show the page on all three potential nav color iterations the-sun (primary-color), dagen (light), and the-prophet (dark) 
5. Should match color effects based on mocks 

<img width="456" alt="Screen Shot 2021-09-27 at 15 01 19" src="https://user-images.githubusercontent.com/5950956/134976893-6a162abe-c6e8-4395-9a72-75a5c64b8108.png">

## Effect Of Changes
### Before
- Dark and light color only 
- Used inconsistent border color 

### After
<img width="1372" alt="Screen Shot 2021-09-27 at 15 03 30" src="https://user-images.githubusercontent.com/5950956/134977231-d0b03c6c-34a2-43d6-8164-bb13e80246bf.png">


## Dependencies or Side Effects
- Shared styles scss has been changed

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [x] Confirmed this PR has unit test files
  - [x] Ran `npm run test`, made sure all tests are passing
  - [x] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored). -> removed theme color params to use properties
- [x] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

Commit:
feat: Add primary filled with button style iterations

Commit:
feat: Add background color primary as well as secondary and reverse outlines

Commit:
feat: Use updated logic for nav color on header action

Commit:
style: Remove unnecessary comment

Commit:
ci: Add grey background for clarity against primary color background

Commit:
feat: Update sign up and login with white background and dark text

Commit:
fix: Use correct helpers for each iteration of identity button

Commit:
style: Use primary naming rather than primary filled for buttons

Commit:
style: Rename white background as primary reverse

Commit:
style: Rename secondary outline reverse to secondary reverse

Commit:
style: Rename secondary outline to secondary

Commit:
refactor: Remove references to outlined

Commit:
refactor: Update naming for primary button and default button theme

Commit:
feat: Add reusable button to nav sections and remove overriding styles

Commit:
feat: Add querly search icon reusable button

Commit:
Merge branch 'canary' into TMEDIA-468-header-nav-chain-buttons

Commit:
test: Update scope of querly search

Commit:
Merge branch 'TMEDIA-468-identity-block-style-button-refactor-pt1' into TMEDIA-468-header-nav-chain-buttons

Commit:
fix: Ensure icontype proptypes match and typo fix for button type

Commit:
feat: Update arc search button based on reusable button

Commit:
refactor: Fix annoying extra space in classname

Commit:
test: Remove unneeded tests

Commit:
test: Remove or update tests as theme color is handled with properties not props

Commit:
test: Remove unused component in tests

Commit:
refactor: Remove unnecessary fallbacks

Commit:
test: Add on click search handler test for querly

Commit:
Merge branch 'TMEDIA-468-identity-block-style-button-refactor-pt1' into TMEDIA-468-header-nav-chain-buttons

Commit:
fix: Add back fallback dark

Commit:
fix: Update color of button for input on show

Commit:
feat: Add custom css for absolutely positioned button

Commit:
fix: Follow zeplin standards for border radius and height

Commit:
fix: Remove newline unnecessary

Commit:
test: Remove unused tests with nav color custom

Commit:
style: Fix newline issue

Commit:
style: Fix stylelint suggestion for order

Commit:
Merge branch 'canary' into TMEDIA-468-header-nav-chain-buttons

Commit:
test: Add functionality tests for shared styles buttons

Commit:
Merge branch 'TMEDIA-468-header-nav-chain-buttons' of github.com:WPMedia/fusion-news-theme-blocks into TMEDIA-468-header-nav-chain-buttons

Commit:
Merge branch 'canary' into TMEDIA-468-header-nav-chain-buttons

matthewroach commented:
Search button is a different size to the menu button - these should be the same

<img width="212" alt="Microsoft Edge-2021-09-28-17-26 21" src="https://user-images.githubusercontent.com/868127/135127404-b51ff55f-c097-485b-9976-5dce48ae9837.png">

Commit:
fix: Standardize heights based on design system

Commit:
fix: Use design system guidance for height of buttons

JackHowa commented:
updated -- thanks @matthewroach -- and wrote sb stories to ensure no changes 

![Screen Shot 2021-09-28 at 13 34 52](https://user-images.githubusercontent.com/5950956/135146180-5d03ee1b-ffae-49c9-950b-9ad786631120.png)

matthewroach commented on blocks/header-nav-chain-block/chains/header-nav-chain-block/_children/search-box.scss:
```diff
@@ -0,0 +1,4 @@
+.search-box--right-absolute-positioned {
```
Looks like this is causing issues when you have the search box configured on the left side

Existing
<img width="197" alt="Microsoft Edge-2021-09-29-11-26 36" src="https://user-images.githubusercontent.com/868127/135253720-ba6acd3e-0c06-4371-b2db-5afdb732eba0.png">

With this change -
<img width="170" alt="Microsoft Edge-2021-09-29-11-26 53" src="https://user-images.githubusercontent.com/868127/135253722-c08c1496-dbb5-4abb-8279-f4c4e7432076.png">

JackHowa replied:
thanks @matthewroach -- updated. only displaying on open now

<img width="184" alt="Screen Shot 2021-09-29 at 09 21 48" src="https://user-images.githubusercontent.com/5950956/135288078-d8d87ff3-375f-481f-9dde-1e79e2c26cf2.png">
<img width="92" alt="Screen Shot 2021-09-29 at 09 21 40" src="https://user-images.githubusercontent.com/5950956/135288085-ca065d3d-4eac-4fef-a522-19460e216130.png">

matthewroach reviewed (changes requested):
Small issue with search when on the left side

Also, I do not see a story for two button side by side?

Commit:
fix: Ensure absolute position only displays on open

Based on feedback https://github.com/WPMedia/fusion-news-theme-blocks/pull/1102#discussion_r718384180

Commit:
docs: Add two buttons to storybook

Commit:
docs: Ensure button side by side

JackHowa commented:
@matthewroach storybook updated with two buttons next to each other 

![image](https://user-images.githubusercontent.com/5950956/135298933-d7d1fac3-5455-47e9-861b-be9052ea5551.png)
